### PR TITLE
Jobs and character slots.

### PR DESCRIPTION
**Pull request by Segrain**
# About the pull request

Works good enough that can probably be put onto a testmerge. Chasing obscure cases was a pain, so something somewhere might still break.

# Explain why it's good for the game

This allows readying up for a wide variety of roles at roundstart without being forced to reuse the same character for all of them. Hopefully it will encourage more players to try having multiple different characters, and even if not, it will at least make life easier for those of us who already do.

# Testing Photographs and Procedure
image

# Changelog
:cl:
add: Jobs can now be assigned specific character slots, which will be automatically loaded when joining as that role.
/:cl:

### CODE REVIEW

**Comment by thwompa** [comment]
we are not hrp

File: code/modules/client/preferences.dm [hunk 1]
```diff
@@ -104,6 +104,8 @@ var/const/MAX_SAVE_SLOTS = 10
 
 	///holds our preferred job options for jobs
 	var/pref_special_job_options = list()
+	///Holds assignment of character slots to jobs.
+	var/pref_job_slots = list()
```
Comment (fira) [inline code/modules/client/preferences.dm:108]:
```suggestion
	var/list/pref_job_slots = list()
```

**Review by fira — COMMENTED** [review]
Code looks good to me but to make sure - this only creates new savfiles, without modifying current ones? If not we might need to hold a proper TM session and backup the save files prior to testmerging to avoid any issues with the decade old mess that our production data is

**Comment by Segrain** [comment]
> Code looks good to me but to make sure - this only creates new savfiles, without modifying current ones? If not we might need to hold a proper TM session and backup the save files prior to testmerging to avoid any issues with the decade old mess that our production data is

Should not be any different from how per-job-preferences were added. If those did not explode savefiles, neither should this.

File: code/__DEFINES/client_prefs.dm [hunk 2]
```diff
@@ -0,0 +1,38 @@
+#define BE_ALIEN_AFTER_DEATH (1<<0)
+#define BE_AGENT (1<<1)
+
+var/list/be_special_flags = list(
```
[the remark this answers is not in the document]
Comment (Segrain) [reply code/__DEFINES/client_prefs.dm:4]: Was as this, I only moved it around. And we do not use agents anyway, but removing that is probably creeping out of scope. But sure, can probably rewrite it a little.

**Review by fira — APPROVED** [review]

**Comment by morrowwolf** [comment]
Testmerged!

**Comment by morrowwolf** [comment]
Code looks good on my end, testing has gone well.

Segrain has stated he wants to make a few tweaks before full merge 🙏

**Pull request merged**